core-models/src/main/scala/com/mesosphere/usi/core/models/faultdomain/DomainSelector.scala

```diff
@@ -0,0 +1,7 @@
+package com.mesosphere.usi.core.models.faultdomain
+
+import org.apache.mesos.v1.Protos.DomainInfo
+
+trait DomainSelector {
```

Comment:
I wonder if this is rather a matcher. And if so maybe we could unify the interfaces for matching. What do you think?

Reply:
It's a special kind of matcher which can take in to consideration the Mesos domain